experimental/emulated_vault/opt/emulated-vault/adapter/fs.py

```diff
@@ -16,298 +16,206 @@
+		:return: A tuple - 'True' for successful retrival and the retrieved value
+		:rtype: Tuple[Boolean, str]
+		"""
+		success, value = self._read_parameter_by_storage_path(self.pingStoragePath)
+		if not success or value is None:
+			self.logger.error("no ping response")
+			return (False, "")
+		self.logger.debug("ping response: %s", value)
+		return (True, value)
+
+	def _read_parameter_by_storage_path(self, parameterStoragePath):
+		"""
+		Reading the value from the provided file name.
+		:param parameterStoragePath: the file name 
+		:type parameterStoragePath: str
+		:return: 'True' for successful retrivaland the retrieved value
+		:rtype: Tuple[boolean, str]
+		"""
+		self.logger.debug("Get parameter by os path: %s", parameterStoragePath)
+		try:
+			with open(parameterStoragePath) as fd:
+				value = fd.read()
+		except Exception as e:
+			self.logger.exception("%s parameter os path not found.", parameterStoragePath)
+			return False, None
+
+		self.logger.debug("Get parameter by os path %s succeed", parameterStoragePath)
+		return True, value
+
+	def _read_parameters_by_storage_path(self, parameterStoragePathPrefix, keyFilters):
+		"""
+		Reading the values of the parameters the provided directory.
+		:param parameterStoragePathPrefix: the directory to look into
+		:type parameterStoragePathPrefix: str
+		:param keyFilters: filter-name/filter-func dict, holding functions that get a key as 
+		input and retunn "true" if key should be included in the result
+		:type keyFilters: Dict[str,function[str]]
+		:return: 'True' for successful retrival and a dict for key-name/value 
+		:rtype: Tuple[boolean, Dict[str, str]]
+		"""
+		self.logger.debug("Get parameters under os path %s", parameterStoragePathPrefix)
+		parameters = {}
+
+		fileNames = []
+		for (dirpath, _, filenames) in os.walk(parameterStoragePathPrefix):
+			fileNames += [os.path.join(dirpath, file) for file in filenames]
+
+		for fileName in fileNames:
+			filteredOut = False
+			for filterName, filterfunc in keyFilters.items():#items() - supporting python 2&3
+				if not filterfunc(fileName):
+					self.logger.debug("Parameter os path %s dropped, not matching filter %s", self._get_parameter_storage_path_from_url_path(fileName), filterName)
+					filteredOut = True
+					break
+			if filteredOut:
+				continue
+
+			parameterUrlPath = self._get_parameter_storage_path_from_url_path(fileName)
+			success, value = self._read_parameter_by_storage_path(fileName)
+			if not success:
+				self.logger.error("%s parameter os path not found.", fileName)
+				return False, None
+				
+			parameters[parameterUrlPath] = value
+
+		return True, parameters
+
+
+	def _write_parameter_by_storage_path(self, parameterStoragePath, value):
+		"""
+		Writing the value to the provided file name.
+		:param parameterStoragePath: the file name 
+		:type parameterStoragePath: str
+		:param value: value to be writen
+		:type value: str
+		:return: 'True' for successful writing
+		:rtype: Boolean		
```

Comment:
You clearly want to support Python 2, so feel free to ignore this (although note it will no longer receive even critical security updates after next month) but if you abandon Python 2 support and use type hinting you can skip all of these `:type` and `:rtype:` tags:

```python3
def _write_parameter_by_storage_path(self, parameterStoragePath: str, value: str) -> bool:
	"""
	Writing the value to the provided file name.
	:param parameterStoragePath: the file name
	:param value: value to be written
	:return: Whether or not the write was successful
	"""
	...
```

Reply:
Ack